x/capability/keeper/keeper.go

```diff
@@ -13,6 +13,8 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/capability/types"
 )
 
+var initialized = false
```

Comment:
There's got to be a cleaner way than to rely on a global :-/

Reply:
I need to find a way to Initialize the memory stores and map on the first transaction that calls GetCapability. I tried first making it a local variable of the ScopedKeeper though this was less efficient since initialization would happen once for each scoped keeper and I would need to change scoped keeper to be a pointer in app.

Reply:
Can you not use a `sync.Once` on the root `keeper`?

Reply:
GetCapability doesn't have access to the root keeper

Reply:
Maybe I could use sync.One as a global variable and then just do it inside `scopedKeeper.GetCapability`

Reply:
Yeah I'm not sure, but relying on a global for critical path code is a strong codesmell and makes testability difficult. We need to find a way that doesn't rely on a global like this IMHO. @robert-zaremba thoughts?

Reply:
Agreed this is a codesmell. We absolutely should have a better fix for 0.35 and I'm confident we can since we can break things.

For just the non-breaking fix tho, maybe I'll write down the issue we're facing and then see if there's a better solution.

The capability store maintains some of its state in the persistent KVStore, and some of its state in a in-memory KVstore and go map.

During regular execution, the capability keeper functions ensure that persistent memstore and memory store are kept in-sync.
In genesis and on app restart, the capability keeper uses state stored on disk to recreate memory store.

With State-sync, the persistent store alone is copied over from other nodes and then fast-sync starts. However unlike genesis and restart, we don't have some trigger point after statesync is complete where we can call into a function and recreate memory store.

So we need to do this lazily, once the first capability is requested. I was originally thinking we could do it only one capability at a time as needed in GetCapability, but that's inefficient because the persistent store doesn't have the right mapping to do that. I think this can be fixed for 0.35. We would need to iterate over all capabilities.

So we want a one-time iteration over all capabilities in persistent store to recreate the memory store, the very first time that `GetCapability` is called.

Reply:
I see. I guess there is no other option huh? I suppose this is OK then 👍